pkg/engine/internal/executor/expressions.go

```diff
@@ -13,7 +13,19 @@ import (
 	"github.com/grafana/loki/v3/pkg/engine/internal/planner/physical"
 )
 
-type expressionEvaluator struct{}
+type expressionEvaluator struct {
+	allocator memory.Allocator
```

Comment:
unwrap can create columns and needs an allocator, so I'm moving it here as I suspect there may be more projections in the future that need an allocator. this also makes it easier to inject in tests, so we can used a checked allocator.